### PR TITLE
Add GitRepository resource interpreter test coverage

### PR DESCRIPTION
**Pull request by Abhay349**
This PR adds missing test coverage for the GitRepository resource interpreter.
It covers status interpretation, health interpretation, dependency detection, retention behavior, and status reflection.

No production logic changes are included.

Thanks to @FAUST-BENCHOU for the valuable reviews and feedback throughout this PR.

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @FAUST-BENCHOU Would you like to review this PR?

**Comment by FAUST-BENCHOU** [comment]
sure let me take a look

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml:1]: renaming to interprethealth-test.yaml is better

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/aggregatestatus-test.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # test case for aggregating status of GitRepository
 # case1. GitRepository with two status items
+# case2. GitRepository with stale member
+# case3. GitRepository with no status items
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/aggregatestatus-test.yaml:7]: consider add cases involving different artifacts (different revisions) (the code only takes the last artifact), and condition merging logic.
https://github.com/karmada-io/karmada/blob/050353ee8fe8d73020db24d32eae7226ad9f9629/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/customizations.yaml#L50-L73

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# test cases for interpreting health of GitRepository
+# case1: GitRepository Ready=True Succeeded should be Healthy
+# case2: GitRepository Ready=True but wrong reason
+# case3: GitRepository Ready=False
+# case4: GitRepository without Ready condition
+# case5: GitRepository with empty conditions
+
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml:9]: consider add nil cases for like `observedObj.status`,`observedObj.status.conditions`

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretstatus-test.yaml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # test case for interpreting status of GitRepository
 # case1. GitRepository: interpret status test
+# case2. GitRepository with nil status
+# case3. GitRepository without resourceTemplateGeneration annotation
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretstatus-test.yaml:9]: consider testing the observedIgnore and observedRecurseSubmodules fields
https://github.com/karmada-io/karmada/blob/050353ee8fe8d73020db24d32eae7226ad9f9629/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/customizations.yaml#L122-L123
The case where the value of the `resourceTemplateGeneration` annotation is not a number.
https://github.com/karmada-io/karmada/blob/050353ee8fe8d73020db24d32eae7226ad9f9629/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/customizations.yaml#L133

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretdependency-test.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# test cases for interpreting dependency of GitRepository
+# case1: GitRepository with secretRef dependency
+# case2: GitRepository with certSecretRef only (not supported)
+# case3: GitRepository with both secretRef and certSecretRef
+# case4: GitRepository with verify.secretRef dependency
+
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretdependency-test.yaml:9]: consider add test case about handling an empty string name like `secretRef.name`,`verify.secretRef.name` and some nil value like `secretRef `,`verify`
The current test only covers `secretRef` and `certSecretRef`, not the combination of `secretRef` and `verify.secretRef`.
Missing tests:
If both `secretRef` and `verify.secretRef` exist and have different names → two dependencies should be returned.
If both `secretRef` and `verify.secretRef` exist and have the same name → only one dependency should be returned (duplicated).

**Comment by FAUST-BENCHOU** [comment]
@XiShanYongYe-Chang @Abhay349 just my opinion, any feeedback is welcome🙃

**Comment by FAUST-BENCHOU** [comment]
part of https://github.com/karmada-io/karmada/issues/6952

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml
Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interprethealth-test.yaml:1]: apologies, let me fix it..

**Comment by Ferryx349** [comment]
Thanks @FAUST-BENCHOU for review!
This detailed feedback would definiately help me a lot....
I will add required missing test cases, let me study the logic again..

Thanks again

**Comment by XiShanYongYe-Chang** [comment]
Thanks to all of you @Abhay349 @FAUST-BENCHOU

**Comment by Ferryx349** [comment]
Really sorry for the delay. I’m traveling back to college today.
Would it be okay if I resume work and share updates by tomorrow EOD?
Thanks :)

**Comment by XiShanYongYe-Chang** [comment]
@Abhay349 It'ok, take it easy.

**Comment by Ferryx349** [comment]
Hi @FAUST-BENCHOU @XiShanYongYe-Chang I have added some tests according to feedback..
Let me know if their are any issues, happy to iterate on further feedback...

Thanks!

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretstatus-test.yaml [hunk 5]
```diff
@@ -45,3 +50,101 @@ observedObj:
...
+  kind: GitRepository
+  metadata:
+    name: sample
+    namespace: test-gitrepository
+    annotations:
+      resourcetemplate.karmada.io/generation: not-a-number
+  status: {}
+operation: InterpretStatus
+output:
+  status: {}
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretstatus-test.yaml:151]: in fact what i mean is the case like 
```
annotations:
  resourcetemplate.karmada.io/generation: "123"
```
to see whether it can works the same as
```
annotations:
  resourcetemplate.karmada.io/generation: 123
```
to prove `tonumber()` works well.However, your test cases are also acceptable.

Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/source.toolkit.fluxcd.io/v1/GitRepository/testdata/interpretstatus-test.yaml:151]: Thanks for the clarification,
I’ve added an explicit test case covering a numeric resourcetemplate.karmada.io/generation value to verify tonumber() behaves consistently for both string and numeric inputs.

**Comment by FAUST-BENCHOU** [comment]
@Abhay349 @XiShanYongYe-Chang lgtm :)

**Comment by XiShanYongYe-Chang** [comment]
> part of #6952

Hi @Abhay349. This information can be directly added to the PR description later.

**Comment by Ferryx349** [comment]
> > part of #6952
> 
> Hi @Abhay349. This information can be directly added to the PR description later.

Sure..

**Comment by Ferryx349** [comment]
@FAUST-BENCHOU @XiShanYongYe-Chang Thanks a lot for review and guidance.
